pixeltable/func/query_template_function.py

```diff
@@ -112,3 +123,85 @@ def make_query_template(py_fn: Callable, param_types: Optional[list[ts.ColumnTyp
     else:
         assert len(args) == 0 and len(kwargs) == 1 and 'param_types' in kwargs
         return lambda py_fn: make_query_template(py_fn, kwargs['param_types'])
+
+
+def retrieval_tool(
```

Comment:
We still have a two-step process here, now? Ie, 
f = pxt.retrieval_tool(...)
tools = pxt.tools(f)

Maybe this should be called retrieval_function(), just to make that clear. Also, you could use it as a udf if you wanted to (and I can see why users would do that).

Maybe discuss those API aspects on pixeltable-dev briefly, just to make sure Jacob and Pierre are happy with it?